### ISSUE TITLE
Battery widget doesn't respect extension_defaults

### ISSUE THREAD

**Issue by OlaRune** [qtile/qtile#3228]
OS: Arch linux
Qtile version: 0.19.0

I have defined `extension_defaults` for all my widgets (see part of  my config file below). This seems to be respected by all the widgets I use except the Battery widget, which seems to respect the foreground color, but not the background color. Manually adding `background=bg` to the Battery widget works.

This line gives a black background (wrong) and the correct foreground color:
`widget.Battery(battery=0, format='{percent:2.0%}', low_percentage=0, show_short_text=False),`

This line gives the correct background color and the correct foreground color:
`widget.Battery(battery=0, format='{percent:2.0%}', low_percentage=0, show_short_text=False, background=bg),`

Part of my config file:
```
# colors
bg = "#282828" 
fg = "#ebdbb2"
gray = "#928374"
bright_gray = "#a89984"
red = "#cc241d"
bright_red = "#fb4934"
green = "#98971a"
bright_green = "#b8bb26"
yellow= "#d79921"
bright_yellow = "#fabd2f"
blue = "#458588"
bright_blue = "#83a598"
purple = "#b16286"
bright_purple = "#d3869b"
aqua = "#689d6a"
faded_aqua = "#8ec07c"

# theme defaults
widget_defaults = dict(
    font='RobotoMono Nerd Font',
    fontsize=16,
    padding=3,
    background=bg,
    foreground=fg
)
extension_defaults = widget_defaults.copy()
```

**Comment by elParaguayo** [comment]
Interesting. I've actually just been looking at a very similar issue for the `Volume` widget (#3203).

I'll take a look.

**Comment by elParaguayo** [comment]
Should be fixed in #3229 if you can test it.
